File: Ruby/TP8/Ej4&9/rectangle.rb
require_relative 'point'
require_relative 'figure'
class Rectangle < Figure
  def initialize(point1, point2)
    @point1=point1 ##izq inferior
    @point2=point2 ##der superior
    @base= (@point1.get_x - @point2.get_x).abs
    @height=( @point1.get_y - @point2.get_y).abs
  end
  def perimeter
    2*@base+2*@height
  end
  def area
    @base*@height
  end
  def to_s
    'Rectangle ['+@point1.to_s+','+@point2.to_s+']'
  end
  def ==(aux)
    aux.instance_of?(Rectangle) && area==aux.area && perimeter==aux.perimeter
  end
end

File: Extras/Parciales/Figures/Rectangle.java
package ar.itba.edu.Parciales.Figures;

public class Rectangle extends Figure {
    Point topLeft;
    Point bottomRight;

    @Override
    public String toString() {
        return String.format("Rectangulo [%s, %s]",topLeft,bottomRight);
    }

    public Rectangle(Point topLeft, Point bottomRight) {
        this.topLeft = topLeft;
        this.bottomRight = bottomRight;
    }

    public double base(){
        return Math.abs(topLeft.getX()-bottomRight.getX());
    }
    public double height(){
        return Math.abs(topLeft.getY()-bottomRight.getY());
    }

    @Override
    double area() {
        return base()*height();
    }

    @Override
    double perimeter() {
        return base()*2+height()*2;
    }

    boolean belongs(Point point) {
        double x = point.getX();
        double y = point.getY();
        return x > topLeft.getX() && x < bottomRight.getX()
                && y > bottomRight.getY() && y < topLeft.getY();
    }
}
